### PR TITLE
add validation on url query params

### PR DESCRIPTION
**Pull request by Omarabdul3ziz**
- separate decoding into filters and limits
- return error if a passed param does not fit on a filter/limit

### Related Issues
- https://github.com/threefoldtech/tfgrid-sdk-go/issues/461

### Checklist

- [x] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstring

### CODE REVIEW

**Review by xmonader — CHANGES_REQUESTED** [review]
I don't understand the PR, please make sure to add comments and tests

File: grid-proxy/internal/explorer/param_parser_test.go [hunk 1]
```diff
@@ -1,74 +1,72 @@
...
-				urlValues.Add(name, fmt.Sprintf("%v", want.Field(i).Index(j).Interface()))
-			}
-		} else {
-			urlValues.Add(name, fmt.Sprintf("%v", y.Interface()))
-		}
+	assert.Equal(t, expectedFilter, gotFilter)
+	assert.Equal(t, expectedLimit, gotLimit)
+}
 
+func TestUnknownParam(t *testing.T) {
```
Comment (xmonader) [inline grid-proxy/internal/explorer/param_parser_test.go:60]: Add another test for a known and unknown param in the same query string

File: grid-proxy/internal/explorer/param_parser.go [hunk 2]
```diff
@@ -1,32 +1,93 @@
...
+
+	return tags
+}
+
+// returns a map of params that have a value
+func ignoreEmptyParams(params url.Values) url.Values {
+	newParams := make(url.Values)
+	for key, val := range params {
+		for _, v := range val {
+			if v != string("") {
```
Comment (xmonader) [inline grid-proxy/internal/explorer/param_parser.go]: why `string("")` instead of `""`?

Comment (Omarabdul3ziz) [reply grid-proxy/internal/explorer/param_parser.go]: i think I was testing something and didn't clean it :D. removed now

**Review by xmonader — APPROVED** [review]

**Pull request merged**